torchrl/objectives/sac.py

```diff
@@ -401,8 +400,21 @@ def target_entropy(self):
                     )
                 if not isinstance(action_spec, CompositeSpec):
                     action_spec = CompositeSpec({self.tensor_keys.action: action_spec})
+                if (
+                    isinstance(self.tensor_keys.action, tuple)
+                    and len(self.tensor_keys.action) > 1
+                ):
+                    action_container_shape = action_spec[
+                        self.tensor_keys.action[:-1]
+                    ].shape
+                else:
+                    action_container_shape = action_spec.shape
                 target_entropy = -float(
-                    np.prod(action_spec[self.tensor_keys.action].shape)
+                    np.prod(
+                        action_spec[self.tensor_keys.action].shape[
+                            len(action_container_shape) :
+                        ]
+                    )
```

Comment:
since we're at it we can do
```suggestion
                    action_spec[self.tensor_keys.action].shape[
                            len(action_container_shape) :
                    ].numel()
```